app/controllers/application_controller/buttons.rb

```diff
@@ -100,6 +100,36 @@ def group_update
   # we need a "Provider" so adding an "ExtManagementSystem" to the list.
   MODEL_WITH_OPEN_URL = %w[ExtManagementSystem Service User MiqGroup Tenant CloudTenant GenericObject Vm].freeze
 
+  # Types of visibility for the custom buttons
+  VISIBILITY_TYPES = {'role' => 'role', 'all' => 'all'}.freeze
+
+  def button_visibility_box_edit
+    typ_changed = params[:visibility_typ].present?
+    @edit[:new][:visibility_typ] = VISIBILITY_TYPES[params[:visibility_typ]] if typ_changed
+
+    visibility_typ = @edit[:new][:visibility_typ]
+    if %w[role].include?(visibility_typ)
+      plural = visibility_typ.pluralize
+      key    = plural.to_sym
+      prefix = "#{plural}_"
+
+      @edit[:new][key] = [] if typ_changed
+      params.each do |var, value|
+        next unless var.starts_with?(prefix)
```

Comment:
do we need to sanitize these params coming in to make sure they're in the right format?

for example, we have `params[:visibility_typ]` above with a symbol but it looks like we're checking if key/var starts_with a prefix, so a string?